python/ray/train/tests/test_state.py

```diff
@@ -155,6 +159,63 @@ def test_state_actor_api():
         assert run_info == info_list[i]
 
 
+class TestConfig(BackendConfig):
+    @property
+    def backend_cls(self):
+        return TestBackend
+
+
+class TestBackend(Backend):
+    def on_start(self, worker_group: WorkerGroup, backend_config: TestConfig):
+        pass
+
+    def on_shutdown(self, worker_group: WorkerGroup, backend_config: TestConfig):
+        pass
+
+
+def test_state_manager(ray_start_gpu_cluster):
+    os.environ["RAY_TRAIN_ENABLE_STATE_TRACKING"] = "0"
```

Comment:
Nit: We don't need this one after removing `BackendExecutor`, but we should use `monkeypatch.setenv(...)` fixture so that the env var is only set within the test.

Otherwise it will spill over to the next one. We should also change the next test's env var setting to use monkeypatch.

Reply:
Ah got it! Good to see that after removing `BackendExecutor`, we don't need to `setenv` anymore.